docs/source/python/orc.rst

```diff
@@ -0,0 +1,261 @@
+                      'two': ['foo', 'bar', 'baz'],
+                      'three': [True, False, True]},
+                      index=list('abc'))
+   df
+   table = pa.Table.from_pandas(df, preserve_index=False)
+
+Then we have:
+
+.. ipython:: python
+
+   po.write_table(table, 'example_noindex.orc')
+   t = po.read_table('example_noindex.orc')
+   t.to_pandas()
+
+Here you see the index did not survive the round trip.
+
+Finer-grained Reading and Writing
+---------------------------------
+
+``read_table`` uses the :class:`~.ORCFile` class, which has other features:
+
+.. ipython:: python
+
+   orc_file = po.ORCFile('example.orc')
+   orc_file.metadata
+   orc_file.schema
+   orc_file.nrows
+
+See the :class:`~pyarrow.orc.ORCFile()` docstring for more details.
+
+As you can learn more in the `Apache ORC format
+<https://orc.apache.org/specification/>`_, an ORC file consists of
+multiple stripes. ``read_table`` will read all of the stripes and
+concatenate them into a single table. You can read individual stripes with
+``read_stripe``:
+
+.. ipython:: python
+
+   orc_file.nstripes
+   orc_file.read_stripe(0)
+
+We can write an ORC file using ``ORCWriter``:
+
+.. ipython:: python
+
+   with po.ORCWriter('example2.orc') as writer:
+      writer.write_table(table)
+
+Compression
+---------------------------------------------
+
+The data pages within a column in a row group can be compressed after the
+encoding passes (dictionary, RLE encoding). In PyArrow we don't use compression
+by default, but Snappy, ZSTD, Gzip/Zlib, and LZ4 are also supported:
+
+.. code-block:: python
+
+   po.write_table(table, where, compression='uncompressed')
+   po.write_table(table, where, compression='gzip')
+   po.write_table(table, where, compression='zstd')
+   po.write_table(table, where, compression='snappy')
+
+Snappy generally results in better performance, while Gzip may yield smaller
+files.
+
+Partitioned Datasets (Multiple Files)
+------------------------------------------------
+
+Multiple ORC files constitute a ORC *dataset*. These may present in a
+number of ways:
+
+* A list of ORC absolute file paths
+* A directory name containing nested directories defining a partitioned dataset
+
+A dataset partitioned by year and month may look like on disk:
+
+.. code-block:: text
+
+   dataset_name/
+     year=2021/
+       month=01/
+          0.orc
+          1.orc
+          ...
+       month=02/
+          0.orc
+          1.orc
+          ...
+       month=03/
+       ...
+     year=2022/
+       month=01/
+       ...
+     ...
+
+
+Reading from cloud storage
+--------------------------
+
+In addition to local files, pyarrow supports other filesystems, such as cloud
+filesystems, through the ``filesystem`` keyword:
+
+.. code-block:: python
+
+    from pyarrow import fs
+
+    s3  = fs.S3FileSystem(region="us-east-2")
+    table = po.read_table("bucket/object/key/prefix", filesystem=s3)
+
+Currently, :class:`HDFS <pyarrow.fs.HadoopFileSystem>` and
```

Comment:
I think we could stop this section here and just redirect the user to the filesystem docs for details.

Reply:
Done!